packages/components/src/DateTimePickers/views/DateTimeView/DateTimeView.component.js

```diff
@@ -6,49 +6,97 @@ import IconButton from '../../IconButton';
 import HeaderTitle from '../HeaderTitle';
 import theme from './DateTimeView.scss';
 
-function DateTimeView(props) {
-	const header = {
-		leftElement: <IconButton
-			icon={{
-				name: 'talend-chevron-left',
-			}}
-			aria-label="Display previous calendar month"
-		/>,
-		middleElement: <HeaderTitle
-			label="Septembre 2017"
-			button={{
-				'aria-label': 'Switch to month and year pickers view',
-			}}
-		/>,
-		rightElement: <IconButton
-			icon={{
-				name: 'talend-chevron-left',
-				transform: 'rotate-180',
-			}}
-			aria-label="Display next calendar month"
-		/>,
-	};
-
-	const bodyElement = (
-		<div className={theme.body}>
-			<div className={theme.date}>
-				<DatePicker />
-			</div>
-			<div className={theme.time}>
+/**
+ * Get the positive euclidean modulo number from a dividend and a divisor
+ * @param {number} a Dividend
+ * @param {number} b Divisor
+ * @return The positive euclidean modulo
+ */
+export function euclideanModulo(a, b) {
+	const m = ((a % b) + b) % b;
+	return m < 0 ? m + Math.abs(b) : m;
+}
+
+class DateTimeView extends React.Component {
+
+	constructor(props) {
+		super(props);
+
+		this.incrementMonthIndexDown = this.incrementMonthIndex.bind(this, -1);
+		this.incrementMonthIndexUp = this.incrementMonthIndex.bind(this, 1);
+	}
+
+	incrementMonthIndex(monthIncrement) {
```

Comment:
Agree with more simple and readable.
But if the increment is changed to more than 1, the code will change. Then if we want more than 12 we'll need a refacto again (2 opportunities to have messy code added). With the actual code, i just don't see cases to need code to grow/be updated.

In my case, yes it takes more time to understand **how** it works, but it can take whatever index increment. And it's a localized _how_ complicated, I think it's as complicated as the problem it's resolving. And after that, the function 'interface' is simple we don't have to know exactly how it's calculated to use it or to understand what the component is achieving.

Reply:
We're dealing with months. So 12 is fixed and the feature is to have buttons to add or substract 1 month.
No need to over-engineer for things that has very very very high probability not to come.

Reply:
I was going to do the 'simple' version, but it seems more complicated after writing it.

**version 1 :**
```javascript
incrementMonthIndex(monthIncrement) {
	const monthIndexIncremented = this.props.monthIndexSelected + monthIncrement;
	const newMonthIndex = euclideanModulo(monthIndexIncremented, 12);
	const yearIncrement = Math.floor(monthIndexIncremented / 12);

	if (this.props.onMonthSelected !== undefined) {
		this.props.onMonthSelected(newMonthIndex);
	}

	if (
		this.props.onYearSelected !== undefined
		&& yearIncrement !== 0
	) {
		const newYear = this.props.yearSelected + yearIncrement;
		this.props.onYearSelected(newYear);
	}
}
```

**version 2 :**
```javascript
incrementMonthIndexOnce(isPositive) {
	const monthIndexIncremented = isPositive
		? this.props.monthIndexSelected + 1
		: this.props.monthIndexSelected - 1;

	let newMonthIndex;
	let newYear;
	if (monthIndexIncremented < 0) {
		newMonthIndex = 11;
		newYear = this.props.yearSelected - 1;
	} else if (monthIndexIncremented > 11) {
		newMonthIndex = 0;
		newYear = this.props.yearSelected + 1;
	} else {
		newMonthIndex = monthIndexIncremented;
	}

	if (this.props.onMonthSelected !== undefined) {
		this.props.onMonthSelected(newMonthIndex);
	}

	if (
		this.props.onYearSelected !== undefined
		&& newYear !== undefined
	) {
		this.props.onYearSelected(newYear);
	}
}
```